File: Random Forest/src/main/java/model/BaselineLearner.java
package model;// ----------------------------------------------------------------
// The contents of this file are distributed under the CC0 license.
// See http://creativecommons.org/publicdomain/zero/1.0/
// ----------------------------------------------------------------

public class BaselineLearner extends SupervisedLearner
{
	private double[] mode;

	public String name()
	{
		return "Baseline";
	}

	public void train(Matrix features, Matrix labels)
	{
		mode = new double[labels.getNumCols()];
		for(int i = 0; i < labels.getNumCols(); i++)
		{
			if(labels.valueCount(i) == 0)
				mode[i] = labels.columnMean(i);
			else
				mode[i] = labels.mostCommonValue(i);
		}
	}

	void predict(double[] sample, double[] prediction)
	{
		Vec.copy(prediction, mode);
	}
}

File: Random Forest/src/main/kotlin/model/learner/BaselineLearner.kt
package model.learner

import model.Matrix
import helpers.Vec.copy

// ----------------------------------------------------------------
// The contents of this file are distributed under the CC0 license.
// See http://creativecommons.org/publicdomain/zero/1.0/
// ----------------------------------------------------------------

class BaselineLearner : SupervisedLearner() {
    private lateinit var mode: DoubleArray

    override val name: String get() = "Baseline"

    override fun train(features: Matrix, labels: Matrix) {
        mode = DoubleArray(labels.cols) { if (labels.valueCount(it) == 0) labels.columnMean(it) else labels.mostCommonValue(it) }
    }

    override fun predict(receiving: DoubleArray, predicting: DoubleArray) = predicting.copy(mode)
}
